services/streammanager/googlecloudfunction/googlecloudfunction.go

```diff
@@ -0,0 +1,223 @@
+package cloudfunctions
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"google.golang.org/api/cloudfunctions/v1"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/idtoken"
+
+	"google.golang.org/api/option"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+	"github.com/tidwall/gjson"
+)
+
+type Config struct {
+	Credentials            string `json:"credentials"`
+	FunctionEnvironment    string `json:"functionEnvironment"`
+	RequireAuthentication  bool   `json:"requireAuthentication"`
+	GoogleCloudFunctionUrl string `json:"googleCloudFunctionUrl"`
```

Comment:
Keep directly function name here instead of url

Reply:
Url is needed in the case of gen2 function.

Reply:
Keep both then

Reply:
Do you mean, I should take both from the user in from UI?

Reply:
Extract function using getFunctionName function and keep it here

Reply:
Done